src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -377,8 +381,15 @@ private void UnregisterRead()
 
             Interop.Error errorCode;
             bool unregistered = _engine.TryRegister(_fileDescriptor, _registeredEvents, events, _handle, out errorCode);
-            Debug.Assert(unregistered, string.Format("UnregisterRead failed: {0}", errorCode));
-            _registeredEvents = events;
+
+            if (unregistered || errorCode == Interop.Error.EBADF)
```

Comment:
We may get this (or a few other errors) in the case that the file descriptor has already been closed.